src/haddock/modules/topology/topocg/aa2cg.py

```diff
@@ -412,43 +431,47 @@ def determine_ss(structure, skipss, pdbf_path):
     """
     # calculate SS
     for model in structure:
+        tmp_file_name = f"../{output_path}/{pdbf_path.split('/')[-1][:-4]}_tmp.pdb"
+        create_file_with_cryst(pdbf_path, tmp_file_name)
+
         if skipss:
             continue
         else:
             try:
-                dssp = DSSP(model, pdbf_path)
+                p = subprocess.Popen(["dssp", tmp_file_name, "--output-format", "dssp"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+                dssp_raw, errors = p.communicate()
+                dssp_raw = dssp_raw.split('#')[1].split('\n')[:-1]
+                if Path(tmp_file_name).exists(): Path(tmp_file_name).unlink()
             except:  # TODO: think about making this exception more specific
                 # no secondary structure detected for this model
+                if Path(tmp_file_name).exists(): Path(tmp_file_name).unlink()
```

Comment:
You don't need to repeat this one, you can put it outside this try/except

Reply:
But if the SS cannot be assigned for whatever reason, the file isn't removed in the folder (because of the continue in the except). So I put it twice. But I can also place it outside of the for loop, before the return.

Reply:
I guess I don't really understand where exaclty you want to put it, since the other options I'm trying are not working if DSSP is failing

Reply:
line 450